File: Cpp/Easy/633. 平方数之和.cpp
给定一个非负整数 c ，你要判断是否存在两个整数 a 和 b，使得 a2 + b2 = c。

示例1:

输入: 5
输出: True
解释: 1 * 1 + 2 * 2 = 5
 

示例2:

输入: 3
输出: False

class Solution {
public:
    bool judgeSquareSum(int c) {
        for(long i=0;i*i<=c;i++){
            double x = sqrt(c-i*i);
            if(x==int(x)){
                return true;
            }
        }
        return false;
    }
};

File: Java/easy/633. 平方数之和.java
class Solution {
    public boolean judgeSquareSum(int c) {
        int i = 0;
        int j = (int)Math.sqrt(c);
        while(i <= j) {
            int powSum = i * i + j * j;
            if(powSum == c) {
                return true;
            } else if(powSum < c) {
                i++;
            } else {
                j--;
            }
        }
        return false;
    }
}
